src/frontend/qt_sdl/LuaMain.cpp

```diff
@@ -0,0 +1,633 @@
+    buttonStartStop->setGeometry(101,0,100,20);
+    buttonOpenScript = new QPushButton("OpenLuaFile",this);
+    buttonOpenScript->setGeometry(202,0,100,20);
+    connect(buttonOpenScript,&QPushButton::clicked,this,&LuaConsoleDialog::onOpenScript);
+    connect(buttonStartStop,&QPushButton::clicked,this,&LuaConsoleDialog::onStop);
+    connect(buttonPausePlay,&QPushButton::clicked,this,&LuaConsoleDialog::onPausePlay);
+    this->setWindowTitle("Lua Script");
+}
+
+void LuaConsoleDialog::closeEvent(QCloseEvent *event)
+{
+    onStop();
+    bundle->overlays->clear();
+    flagClosed = true;
+    event->accept();
+}
+
+void LuaConsoleDialog::onOpenScript()
+{
+    QFileInfo file = QFileInfo(QFileDialog::getOpenFileName(this, "Load Lua Script",QDir::currentPath()));
+    if (!file.exists()) return;
+    currentScript = file;
+    bundle->flagNewLua = true;
+}
+
+LuaConsole::LuaConsole(QWidget* parent)
+{
+    this->setParent(parent);
+}
+
+void LuaConsole::onGetText(const QString& string)
+{
+    this->appendPlainText(string);
+    QScrollBar* bar = verticalScrollBar();
+    bar->setValue(bar->maximum());
+}
+
+void LuaBundle::printText(QString string)
+{
+    this->luaDialog->console->onGetText(string);
+}
+
+void LuaConsoleDialog::onLuaSaveState(QString string)
+{
+    emit signalLuaSaveState(string);
+}
+
+void LuaConsoleDialog::onLuaLoadState(QString string)
+{
+    emit signalLuaLoadState(string);
+}
+
+void LuaConsole::onClear()
+{
+    this->clear();
+}
+
+LuaFunction::LuaFunction(luaFunctionPointer cf,const char* n,std::vector<LuaFunction*>* container)
+{
+    this->cfunction = cf;
+    this->name = n;
+    container->push_back(this);
+}
+
+static_assert(sizeof(LuaBundle*) <= LUA_EXTRASPACE,"LUA_EXTRASPACE too small");
+
+LuaBundle* get_bundle(lua_State * L)
+{
+	LuaBundle* pBundle;
+	std::memcpy(&pBundle, lua_getextraspace(L), sizeof(LuaBundle*));
+	return pBundle;
+}
+
+#define MELON_LUA_HOOK_INSTRUCTION_COUNT 50 //number of vm instructions between hook calls
+void luaHookFunction(lua_State* L, lua_Debug *arg)
+{
+    LuaBundle* bundle = get_bundle(L);
+    if (bundle->flagStop and (arg->event == LUA_HOOKCOUNT)) 
+        luaL_error(L, "Force Stopped");
+}
+
+std::vector<LuaFunction*> definedLuaFunctions;//List of all defined lua functions
+
+void LuaBundle::createLuaState()
+{
+    if (!flagNewLua) return;
+    overlays->clear();
+    flagNewLua = false;
+    luaState = nullptr;
+    std::string fileName = luaDialog->currentScript.fileName().toStdString();
+    std::string filedir = luaDialog->currentScript.dir().path().toStdString();
+    lua_State* L = luaL_newstate();
+    LuaBundle* pBundle = this;
+    std::memcpy(lua_getextraspace(L), &pBundle, sizeof(LuaBundle*)); //Write a pointer to this LuaBundle into the extra space of the new lua_State
+    luaL_openlibs(L);
+    for (LuaFunction* function : definedLuaFunctions)
+        lua_register(L,function->name,function->cfunction);
+    std::filesystem::current_path(filedir.c_str());
```

Comment:
I am quite sure this is problematic and leads to issues on Windows with non ASCII characters. Either use wide strings or std::filesystem::u8path.